### PR TITLE
14.0 mig product form sale link

### PR DESCRIPTION
**Pull request by dani-xtendoo**
Product form sale link migration to 14.0

### CODE REVIEW

**Review by ajaniszewska-dev — APPROVED** [review]

File: product_form_sale_link/__manifest__.py [hunk 1]
```diff
@@ -1,14 +1,15 @@
+# Copyright 2021 Xtendoo
```
Comment (rousseldenis) [inline product_form_sale_link/__manifest__.py]: Don't add yourself to authors for a migration.

You should add just a line in CONTRIBUTORS

File: product_form_sale_link/__manifest__.py [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
 {
     "name": "Product Form Sale Link",
     "summary": """
         Adds a button on product forms to access Sale Lines""",
-    "version": "13.0.1.0.1",
+    "version": "14.0.1.0.1",
     "license": "AGPL-3",
     "category": "Sales",
-    "author": "ACSONE SA/NV,Odoo Community Association (OCA)",
+    "author": "Xtendoo, ACSONE SA/NV,Odoo Community Association (OCA)",
```
Comment (rousseldenis) [inline product_form_sale_link/__manifest__.py]: Same as above

File: product_form_sale_link/__manifest__.py [hunk 3]
```diff
@@ -1,14 +1,15 @@
+# Copyright 2021 Xtendoo
 # Copyright 2019 ACSONE SA/NV
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 {
     "name": "Product Form Sale Link",
     "summary": """
         Adds a button on product forms to access Sale Lines""",
-    "version": "13.0.1.0.1",
+    "version": "14.0.1.0.1",
```
Comment (rousseldenis) [inline product_form_sale_link/__manifest__.py]: You need to reset this to 14.0.1.0.0

File: product_form_sale_link/views/product_product.xml [hunk 4]
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8" ?>
-<!-- Copyright 2019 ACSONE SA/NV
-     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<!-- Copyright 2021 Xtendoo
```
Comment (rousseldenis) [inline product_form_sale_link/views/product_product.xml]: Same as above

File: product_form_sale_link/views/product_template.xml [hunk 5]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="utf-8" ?>
-<!-- Copyright 2019 ACSONE SA/NV
+<!-- Copyright 2021 Xtendoo
```
Comment (rousseldenis) [inline product_form_sale_link/views/product_template.xml]: Same as above

File: product_form_sale_link/views/sale_order_line.xml [hunk 6]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="utf-8" ?>
-<!-- Copyright 2019 ACSONE SA/NV
+<!-- Copyright 2021 Xtendoo
```
Comment (rousseldenis) [inline product_form_sale_link/views/sale_order_line.xml]: Same as above

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Hi @dani-xtendoo 

Thanks for your contribution.

Could you read https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0 ?

Especially the 'Full process' part ? https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0#howto

You need to have one commit for pre-commit and one for the migration itself.

**Review by manuelcalerosolis — APPROVED** [review]
Functional and code review!

**Comment by colmeirin** [comment]
There is a bit of an issue with the module. It works OK with only "sale" module installed but if you have installed "stock" module, it crashes the installation with the view product_template.xml. This happens because odoo core "stock" module has a view that deletes the button name="action_view_sales" by replacing it(id="product_template_view_form_inherit_sale") and redefines it in another view(id="product_template_view_form_inherit_stock") so this module can't find the button to inherit and therefore crashes the instalation.

**Comment by rousseldenis** [comment]
Closing this as already migrated in #2012

**Pull request closed without merging**